src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOverridingAndHiding.cs

```diff
@@ -1112,54 +1112,156 @@ public static void Main()
+                        switch (i)
+                        {
+                            case 0:
+                                compilation.VerifyDiagnostics(
+                                    // (5,25): warning CS1957: Member 'Derived.Method(ref List<int>, out List<int>)' overrides 'Base<int, int>.Method(ref List<int>, out List<int>)'. There are multiple override candidates at run-time. It is implementation dependent which method will be called. Please use a newer runtime.
+                                    //     public virtual void Method(ref List<T> x, out List<U> y) { y = null; Console.WriteLine("Base<T, U>.Method(ref List<T> x, out List<U> y)"); }
+                                    Diagnostic(ErrorCode.WRN_MultipleRuntimeOverrideMatches, "Method").WithArguments("Base<int, int>.Method(ref System.Collections.Generic.List<int>, out System.Collections.Generic.List<int>)", "Derived.Method(ref System.Collections.Generic.List<int>, out System.Collections.Generic.List<int>)").WithLocation(5, 25)
+                                    );
+                                break;
+                            default:
+                                compilation.VerifyDiagnostics(
+                                    // (6,25): warning CS1957: Member 'Derived.Method(ref List<int>, out List<int>)' overrides 'Base<int, int>.Method(ref List<int>, out List<int>)'. There are multiple override candidates at run-time. It is implementation dependent which method will be called. Please use a newer runtime.
+                                    //     public virtual void Method(ref List<T> x, out List<U> y) { y = null; Console.WriteLine("Base<T, U>.Method(ref List<T> x, out List<U> y)"); }
+                                    Diagnostic(ErrorCode.WRN_MultipleRuntimeOverrideMatches, "Method").WithArguments("Base<int, int>.Method(ref System.Collections.Generic.List<int>, out System.Collections.Generic.List<int>)", "Derived.Method(ref System.Collections.Generic.List<int>, out System.Collections.Generic.List<int>)").WithLocation(6, 25)
+                                    );
+                                break;
+                        }
+                        var (m1, m2) = i switch
+                        {
+                            0 => ("Base<T, U>.Method(ref List<T> x, out List<U> y)", "Derived.Method(ref List<int> a, out List<int> b)"),
+                            _ => ("Derived.Method(ref List<int> a, out List<int> b)", "Base<T, U>.Method(out List<U> y, ref List<T> x)"),
+                        };
+                        expectedOutput = $@"
+{m1}
+{m2}
+Derived.Method(BASEREF List<int> a)
+Base2<A, B>.Method(BASE2REF List<A> x)
+
+{m1}
+{m2}
+Derived.Method(BASEREF List<int> a)
+Base2<A, B>.Method(BASE2REF List<A> x)
+
+{m1}
+{m2}
+Derived.Method(BASEREF List<int> a)";
+                    }
+                    else
+                    {
+                        // unknown runtime behavior
+                        continue;
```

Comment:
>continue; [](start = 24, length = 9)

Can't we test Mono behavior today?

Reply:
Not in our unit tests. There are some issues getting their runtime working in our repo. I will keep that in mind as I'm working to get dotnet5 inserted into our repo. Mono working again will be next on my list after that.

Reply:
> Not in our unit tests. There are some issues getting their runtime working in our repo. I will keep that in mind as I'm working to get dotnet5 inserted into our repo. Mono working again will be next on my list after that. 

Perhaps it would make sense to make a note to come back to this place once that happens?

---
In reply to: [446446062](https://github.com/dotnet/roslyn/pull/44025#discussion_r446446062) [](ancestors = 446446062)

Reply:
In `ConditionalFactAttribute` we have
```csharp
        public static bool IsDesktop => RuntimeUtilities.IsDesktopRuntime;
        public static bool IsCoreClr => !IsDesktop;
```
So `mono` will be tested as a core clr runtime.

---
In reply to: [446445205](https://github.com/dotnet/roslyn/pull/44025#discussion_r446445205) [](ancestors = 446445205)

Reply:
Mono qualifies as a desktop runtime.

Reply:
In any case, mono would be tested in the `(ExecutionConditionUtil.IsCoreClr || ExecutionConditionUtil.IsDesktop)` scenario of line 1213

---
In reply to: [446455428](https://github.com/dotnet/roslyn/pull/44025#discussion_r446455428) [](ancestors = 446455428)